salt/transport/tcp.py

```diff
@@ -82,11 +82,9 @@ def __new__(cls, opts, **kwargs):
 
     @classmethod
     def __key(cls, opts, **kwargs):
-        if 'master_uri' in kwargs:
-            opts['master_uri'] = kwargs['master_uri']
         return (opts['pki_dir'],     # where the keys are stored
                 opts['id'],          # minion ID
-                opts['master_uri'],  # master ID
+                kwargs.get('master_uri', opts.get('master_uri')),  # master ID
```

Comment:
if `master_uri` in `kwargs` is set to `None` the value from `opts` won't be evaluated...
You might be better with:
```python
args.get('master_uri') or opts.get('master_uri')),
```
```python
>>> a = {'foo': None}
>>> b = {'foo': 1}
>>> a.get('foo', b.get('foo'))
>>> a.get('foo') or b.get('foo')
1
>>> 
```